sql/parser/eval.go

```diff
@@ -134,15 +137,23 @@ var binOps = map[binArgs]binOp{
 		},
 	},
 	binArgs{Plus, dateType, intervalType}: {
-		returnType: DummyTimestamp,
+		returnType: DummyDate,
 		fn: func(left Datum, right Datum) (Datum, error) {
-			return DTimestamp{Time: left.(DDate).Add(right.(DInterval).Duration)}, nil
+			days, err := inDays(right.(DInterval).Duration)
+			if err != nil {
+				return DummyDate, err
+			}
+			return DDate(left.(DDate) + days), nil
```

Comment:
@vivekmenezes's point (with this change; I also don't understand "forget about intervals") is that adding an interval to a `DATE` is a kind of implicit cast (unless that interval is an integer number of days), just like adding an int and a float. It would be more consistent with our stance on implicit casts elsewhere to require the date to be explicitly cast to a timestamp (even though it is inconsistent with postgres)